lib/packwerk/node_processor_factory.rb

```diff
@@ -10,14 +10,13 @@ class NodeProcessorFactory < T::Struct
 
     const :root_path, String
     const :reference_lister, ReferenceLister
-    const :node_processor_class, T.class_of(NodeProcessor)
     const :context_provider, Packwerk::ConstantDiscovery
     const :constant_name_inspectors, T::Array[ConstantNameInspector]
     const :checkers, T::Array[Checker]
 
     sig { params(filename: String, node: AST::Node).returns(NodeProcessor) }
     def for(filename:, node:)
-      node_processor_class.new(
+      ::Packwerk::NodeProcessor.new(
```

Comment:
This is good I think. I find it easier to understand this way, and if we’re not losing any capability then that seems like a win. As always, we can easily switch back to injecting the class if/when we ever need it.